pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -1241,6 +1397,76 @@ func (m *kubeGenericRuntimeManager) SyncPodContainer(pod *v1.Pod, podStatus *kub
 	return
 }
 
+func (m *kubeGenericRuntimeManager) updateContainerLimits(pod *v1.Pod, podStatus *kubecontainer.PodStatus, resourceName v1.ResourceName, containersToUpdate []containerToUpdateInfo) error {
+	var updatedLimitsMap map[kubecontainer.ContainerID]struct {
+		cStatus *kubecontainer.ContainerStatus
+		rLimits v1.ResourceList
+	}
+	updatedLimitsMap = make(map[kubecontainer.ContainerID]struct {
+		cStatus *kubecontainer.ContainerStatus
+		rLimits v1.ResourceList
+	})
+
+	for _, cInfo := range containersToUpdate {
+		if *cInfo.restart == true {
+			continue
+		}
+		container := &pod.Spec.Containers[cInfo.specIndex]
+		specLimits := container.Resources.Limits
+		defer func() {
+			container.Resources.Limits = specLimits
```

Comment:
this is to revert in case of failure right?  please consider add a comment here.

Reply:
yes.

Reply:
this is actually to revert when exiting the function - appears to be a standard golang pattern, not just k8s pattern.